index.js

```diff
@@ -6,46 +6,55 @@
 'use strict';
 module.exports = leftPad;
 
-var cache = [
-  '',
-  ' ',
-  '  ',
-  '   ',
-  '    ',
-  '     ',
-  '      ',
-  '       ',
-  '        ',
-  '         '
-];
+// Generate cache. Strings are UTF16 in JavaScript,
+// so this will be (1 + 2 + … + 32) = 528*2 bytes,
+// or about 1KiB of cache per array (plus a little
+// bit extra overhead). 
+// Probably not a significant amount.
+var cache = ['',' '];
+for (var i = 2; i <= 32; i++){
+  cache.push(cache[i-1]+' ');
+}
 
 function leftPad (str, len, ch) {
-  // convert `str` to a `string`
-  str = str + '';
+  // initialise `pad` as string early - more monomorphic and helps with str coercion
+  var pad = '';
+  // convert `str` to a `string` via type coercion
+  str = str + pad;
   // `len` is the `pad`'s length now
   len = len - str.length;
   // doesn't need to pad
   if (len <= 0) return str;
-  // `ch` defaults to `' '`
+  // `ch` defaults to `' '` otherwise
+  // convert `ch` to a `string` since 
+  // it could be a number
   if (!ch && ch !== 0) ch = ' ';
-  // convert `ch` to a `string` cuz it could be a number
-  ch = ch + '';
-  // cache common use cases
-  if (ch === ' ' && len < 10) return cache[len] + str;
-  // `pad` starts with an empty string
-  var pad = '';
-  // loop
+  else ch = ch + pad; 
+  // common use cases
+  if (ch === ' ') {
+    // if less or equal to 32 we have a ready-made padding
+    if (len <= 32) return cache[len] + str;
+    // initialise pad at correct size and
+    // skip the first five iterations
+    // of the loop below
+    pad = cache[len & 31];
+    len >>= 5;
+    ch = cache[32];
+  }
   while (true) {
     // add `ch` to `pad` if `len` is odd
-    if (len & 1) pad += ch;
+    if (len & 1) pad = ch + pad;
     // divide `len` by 2, ditch the remainder
-    len >>= 1;
-    // "double" the `ch` so this operation count grows logarithmically on `len`
-    // each time `ch` is "doubled", the `len` would need to be "doubled" too
-    // similar to finding a value in binary search tree, hence O(log(n))
-    if (len) ch += ch;
-    // `len` is 0, exit the loop
-    else break;
+    if (len >>= 1) {
+      // "double" the `ch` so this operation count grows logarithmically on `len`
+      // each time `ch` is "doubled", the `len` would need to be "doubled" too
+      // similar to finding a value in binary search tree, hence O(log(n))
+      ch += ch;
+      // only reached is `len` is not zero, so we can restart loop early
```

Comment:
"if"

Reply:
I don't see what you mean? You think there is a missing if? It's right there at line 48

Reply:
I meant the text, "only reached **is** `len` is not zero"

Reply:
Ahaha! Good catch. Sorry, I wrote this around... like... 5 AM after a night of hyperfocused trial-and-error-programming and forgetting the need to sleep (so this definitely needs some more bug-testing too)

Reply:
Well, thanks for picking this up - if you're interested in putting "debug, optimize, repeat" skills into use - we could also use an extra hand at Node improving stuff :)

Reply:
I'll be looking for part-time freelancing work starting in May, actually

Reply:
I can't pay you for Node (I don't get paid either) but the problems are really interesting https://github.com/nodejs/node/blob/master/CONTRIBUTING.md :)

Reply:
Ah, like that! Well, if I have time :) 
So many fun projects out there that I would love to tinker with...